### ISSUE TITLE
Add support for reading in coils from make_grid type files

### ISSUE THREAD

**Issue by dpanici** [PlasmaControl/DESC#460]
codes like REGCOIL output `coil.XXX` text files in the format

```
X_1  Y_1 Z_1 Current
X_2  Y_2 Z_2 Current
...
X_nseg  Y_nseg Z_nseg Current
(repeated for each coil in the coilset)
```
It would be useful to be able to interface with these outputs files
- [x] Add methods to `FourierXYZCoil` to read in and fit the curves from these files
- [x] Add new coil class `XYZCoil` to be able to just store the coil segment locations in space (without needing to Fourier Transform)

**Comment by f0uriest** [comment]
Do you have an example file? I was meaning to add spline coils which would be ideal for this

**Comment by dpanici** [comment]
example coil file from REGCOIL, which FOCUS is able to read so seems to be a standard format




[coils.regcoil_out.LM_7__N_7__WQS_1.1__alpha_1000__QSCIND_26qsc_26_opt7_Ncoils_120.txt](https://github.com/PlasmaControl/DESC/files/11092272/coils.regcoil_out.LM_7__N_7__WQS_1.1__alpha_1000__QSCIND_26qsc_26_opt7_Ncoils_120.txt)

**Comment by f0uriest** [comment]
Is this resolved by #494 ?

**Comment by dpanici** [comment]
More or less, the FourierXYZCoil can be made by reading in a SplineXYZCoil and then converting it with from_values so I'd say yes
